storage/aptosdb/src/db/include/aptosdb_internal.rs

```diff
@@ -85,6 +85,23 @@ impl AptosDB {
             max_num_nodes_per_lru_cache_shard,
         )?;
 
+        // check if internal indexer db and main db is violating contraint:
+        // when internal statekey indexer db is empty and kv restore progress should also be empty
+        if internal_indexer_db.is_some() {
+            let internal_db: &InternalIndexerDB = internal_indexer_db
+                .as_ref()
+                .expect("InternalIndexerDB should be Some");
+            if internal_db.statekeys_enabled() {
+                let version = internal_db.get_persisted_version()?;
+                let restore_progress = ledger_db.get_in_progress_state_kv_snapshot_version()?;
+                let missing_state_keys = version == 0 && restore_progress.is_some();
+                ensure!(
+                    !missing_state_keys,
+                    "Internal indexer DB is empty but restore progress is not empty."
+                );
+            }
```

Comment:
this will prevent the retore tool to resume after interrupted in the middle of a snapshot recovery?